### PR TITLE
PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by Geno419**

### CODE REVIEW

File: __tests__/test.js [hunk 1]
```diff
@@ -92,6 +93,40 @@ describe("GET /api/articles ", () => {
   });
 });
 
+describe("GET /api/articles/:article_id/comments ", () => {
+  test(".../:article_id/comments returns all comments for given article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .then((res) => {
+        expect(res.status).toBe(200);
+        expect(res.body.length).toBeGreaterThan(0);
```
Comment (nc-richard-neat) [inline __tests__/test.js:102]: Can we be more specific? We know what the data looks like 👍

File: __tests__/test.js [hunk 2]
```diff
@@ -92,6 +93,40 @@ describe("GET /api/articles ", () => {
...
+  test("returns 404 when article_id does not exist", () => {
+    const article_id = "888";
+    return request(app)
+      .get(`/api/articles/${article_id}/comments`)
+      .then((err) => {
+        expect(err.status).toBe(404);
+        expect(err.text).toBe(`${article_id} is not an article`);
+      });
+  });
+});
```
Comment (nc-richard-neat) [inline __tests__/test.js:128]: A 400 invalid article_id test would be useful here too 👍

File: db/model/model.js [hunk 3]
```diff
@@ -112,17 +97,36 @@ exports.fetchPostedComment = (article_id, username, body) => {
     });
 };
 
-exports.verifyDetails = (article_id, username, res, next) => {
+exports.updateVoteByArticleId = (IncrementBy, article_id, res, next) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .query(
```
Comment (nc-richard-neat) [inline db/model/model.js:102]: No need to select the article first, lets run the update query and for error handling allow psql to throw an error if it's going to, or look at the response if it isn't. A checkExists query is a last resort if we can't get an error any other way.

File: db/model/model.js [hunk 4]
```diff
@@ -112,17 +97,36 @@ exports.fetchPostedComment = (article_id, username, body) => {
...
+        SET votes = $1
+        WHERE article_id = $2
+        RETURNING *;`,
+        [String(updatedVote), article_id]
+      );
     })
+    .then(({ rows }) => {
+      return rows[0];
+    })
+    .catch((err) => {
```
Comment (nc-richard-neat) [inline db/model/model.js:123]: We should only have a .catch block at the end of our promise chain which in this case is our controller, you may need to look at your other models too to remove all unnecessary catch blocks.

File: __tests__/test.js (hunk 1 above)
Comment (nc-poonam-rajput) [reply __tests__/test.js:102]: Again, can you assert that the object that comes back is a article that has had it's votes changed?

**Pull request merged**